File: lc169.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        int majority = nums[0];
        int count=1;
        for(int i=1;i<nums.Length;i++)
        {
            if(majority == nums[i])
            {
                count++;
            }
            else
            {
                count--;
                if(count<=0)
                {
                    count = 1;
                    majority = nums[i];
                }
            }
        }
        return majority;
    }
}

File: python/lc169.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        d = {}
        majority = 0
        max_count = 0
        for n in nums:
            if n in d:
                d[n] += 1
            else:
                d[n] = 1
            if max_count < d[n]:
                max_count = d[n]
                majority = n
        return majority
